### PR TITLE
Take screenshots on build agents during tests

### PR DESCRIPTION
**Pull request by RussKie**
## Proposed changes

- While tests are running capture screenshots every 3 minutes to try to gain visibility as to why tests are currently stalling.

image

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/winforms/pull/3056)

### CODE REVIEW

**Comment by weltkante** [comment]
Its been a while but [this](https://github.com/dotnet/winforms/issues/2192#issuecomment-546295892) never got addressed I think:

> [...] there are a whole bunch of forms open even though no other test is running any more [...] These unresponsive forms are associated with threads which are not UI threads and are not running a message loop [...]

So there are probably still tests around which don't dispose their forms and/or aren't annotated as requiring a UI thread (unless they already got fixed and #2192 was forgotten to be reenabled). Just mentioning because the "taking screenshots" PR reminded me that the screen never was clean in the first place.

File: eng/Screenshots.win.psm1 [hunk 1]
```diff
@@ -0,0 +1,167 @@
+# Collection of powershell build utility functions that we use across our scripts.
```
[the remark this answers is not in the document]
Comment (RussKie) [reply eng/Screenshots.win.psm1:1]: Thank you, missed it!

[the remark this answers is not in the document]
Comment (RussKie) [reply eng/Screenshots.win.psm1:1]: Added, though doesn't look like any of our PS scripts have the licensing comments.

**Comment by RussKie** [comment]
Thank you @weltkante. 
There could be multiple reasons. E.g. we still have some tests with incorrect decorations (e.g. `Fact` instead of `WinFormsFact`). There could also be failures resulting in blocking popups, and this work is a stepping stone in attempt to ascertain there aren't anything else in play.
Some race conditions may have been addressed by utilising `RemoteExecutor`, but there obviously more work to be done.

**Comment by RussKie** [comment]
Thank you @weltkante. 
There could be multiple reasons. 
There could be failures resulting in blocking popups, and this work is an attempt to verify that.
We also still have some tests with incorrect decorations (e.g. `Fact` instead of `WinFormsFact`).  Some race conditions may have been addressed by utilising `RemoteExecutor`, but there obviously more work to be done.

**Pull request merged**